pkg/util/file_indexer_test.go

```diff
@@ -0,0 +1,129 @@
+package util
+
+import (
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/openshift/odo/pkg/testingutil/filesystem"
+)
+
+func TestCheckGitIgnoreFile(t *testing.T) {
+
+	// create a fake fs in memory
+	fs := filesystem.NewFakeFs()
+	// create a context directory on fake fs
+	contextDir, err := fs.TempDir(os.TempDir(), "context")
+	if err != nil {
+		t.Error(err)
+	}
+
+	gitignorePath := filepath.Join(contextDir, ".gitignore")
+
+	tests := []struct {
+		testName        string
+		create          bool
+		gitIgnoreCreate func(create bool, contextDir string, fs filesystem.Filesystem) error
+		directory       string
+		want            string
+		wantErr         bool
+	}{
+		{
+			testName:        "Test when .gitignore does not exist",
+			create:          true,
+			gitIgnoreCreate: mockDirectoryInfo,
+			directory:       contextDir,
+			want:            gitignorePath,
+			wantErr:         false,
+		},
+		{
+			testName:        "Test when .gitignore exists",
+			create:          false,
+			gitIgnoreCreate: mockDirectoryInfo,
+			directory:       contextDir,
+			want:            gitignorePath,
+			wantErr:         false,
+		},
+	}
+
+	for _, tt := range tests {
+
+		err := tt.gitIgnoreCreate(tt.create, tt.directory, fs)
+		if err != nil {
+			t.Error(err)
+		}
+
+		t.Run(tt.testName, func(t *testing.T) {
+
+			gitIgnoreFilePath, err := checkGitIgnoreFile(tt.directory, fs)
+
+			if tt.want != gitIgnoreFilePath {
+				t.Errorf("checkGitIgnoreFile unexpected error %v, while creating .gitignore file", err)
+			}
+
+			if !tt.wantErr == (err != nil) {
+				t.Errorf("checkGitIgnoreFile unexpected error %v, wantErr %v", err, tt.wantErr)
+			}
+		})
+	}
+
+}
+
+func TestAddOdoFileIndex(t *testing.T) {
+
+	// create a fake fs in memory
+	fs := filesystem.NewFakeFs()
+	// create a context directory on fake fs
+	contextDir, err := fs.TempDir(os.TempDir(), "context")
+	if err != nil {
+		t.Error(err)
+	}
+
+	gitignorePath := filepath.Join(contextDir, ".gitignore")
+
+	tests := []struct {
+		testName        string
```

Comment:
Same as earlier. We use `name`.

Reply:
https://github.com/openshift/odo/pull/2441/files#r358119130